videoanalyst/model/module_base.py

```diff
@@ -41,6 +46,36 @@ def set_hps(self, hps: dict()) -> None:
             self._hyper_params[key] = hps[key]
 
     def update_params(self):
-        r"""
-        an interface for update params
-        """
+        model_file = self._hyper_params.get("pretrain_model_path", "")
+        if model_file != "":
+            state_dict = torch.load(model_file,
+                                    map_location=torch.device("cpu"))
+            if "model_state_dict" in state_dict:
+                state_dict = state_dict["model_state_dict"]
+            self.load_model_param(state_dict)
+            logger.info(
+                "Load pretrained {} parameters from: {} whose md5sum is {}".
+                format(self.__class__.__name__, model_file, md5sum(model_file)))
+
+    def load_model_param(self, checkpoint_state_dict):
```

Comment:
Where did these lines of code come from (just to confirm its source)?

Reply:
detectron